src/main/java/org/example/gabartischolarsqbbe/controller/AuthController.java

```diff
@@ -0,0 +1,82 @@
+package org.example.gabartischolarsqbbe.controller;
+
+import org.example.gabartischolarsqbbe.dto.LoginRequest;
+import org.example.gabartischolarsqbbe.dto.SignupRequest;
+import org.example.gabartischolarsqbbe.entity.User;
+import org.example.gabartischolarsqbbe.service.AuthService;
+import org.example.gabartischolarsqbbe.config.JwtUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    @Autowired
+    private AuthService authService;
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> registerUser(@RequestBody SignupRequest signupRequest) {
```

Comment:
I think we should have some standardization between the endpoint and the method name, so maybe rename the endpoint to `\register`